java/src/main/native/ai_onnxruntime_OnnxRuntime.c

```diff
@@ -76,8 +76,13 @@ JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OnnxRuntime_getAvailableProvi
 JNIEXPORT jstring JNICALL Java_ai_onnxruntime_OnnxRuntime_initialiseVersion
   (JNIEnv * jniEnv, jclass clazz) {
   (void)clazz;  // required JNI parameter not needed by functions which don't access their host class.
-  const char* version = OrtGetApiBase()->GetVersionString();
+  const ORTCHAR_T* version = OrtGetApiBase()->GetVersionString();
   assert(version != NULL);
+#ifdef _WIN32
+  size_t len = wcslen(version);
+  jstring versionStr = (*jniEnv)->NewString(jniEnv, (const jchar*)version, len);
```

Comment:
If I remember correctly, Java use UTF-16-BE while most Windows use UTF-16-LE.  @Craigacp, is it true?

Reply:
We don't transform the output of jstring->GetStringChars to change UTF-16 endianness on Windows, and it can load & save things to those paths correctly. So I would expect it to be the same on platforms that we currently test in the CI. If it's the same on all platforms is beyond my knowledge of Windows. It should be straightforward to modify https://github.com/microsoft/onnxruntime/blob/main/java/src/test/java/ai/onnxruntime/InferenceTest.java#L74 to test that the output is correct. I haven't modified that test to check the string because it'll need updating every time the version number is bumped so it seems like it would fail a bunch of pipelines.